drv/stm32h7-update-server/src/main.rs

```diff
@@ -421,6 +422,100 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
         // TODO(AJS): Return actual info about loaded images for the SP
         Ok(UpdateStatus::Sp)
     }
+
+    fn read_image_caboose(
+        &mut self,
+        _: &RecvMessage,
+        name: [u8; 4],
+        data: Leased<idol_runtime::W, [u8]>,
+    ) -> Result<u32, RequestError<CabooseError>> {
+        // This code is very similar to `kipc::read_caboose_pos`, but it
+        // operates on the alternate flash bank rather than on the loaded image.
+        let image_start = unsafe { __REGION_BANK2_BASE.as_ptr() } as u32;
+
+        // If all is going according to plan, there will be a valid Hubris image
+        // flashed into the other slot, which is delimited by
+        // `__REGION_BANK2_BASE` and `__REGION_BASE2_END` (which are symbols
+        // injected by the linker).
+        //
+        // We'll first want to read the image header, which is at a fixed
+        // location at the end of the vector table.
+        const HEADER_OFFSET: u32 = 0x298;
```

Comment:
Is this offset determined by our build system or chip hardware or something else?

Reply:
A little from column A, a little from column B: our build system places the `ImageHeader` directly after the vector table, which is determined by the chip hardware.  I added a little more detail to this comment.